File: src/com/ikov/commands/ranks/ModeratorCommands.java
package com.ikov.commands.ranks;

import com.ikov.commands.CommandHandler;
import com.ikov.commands.ModeratorCommand;
import com.ikov.model.Locations.Location;
import com.ikov.model.Position;
import com.ikov.world.World;
import com.ikov.world.content.PlayerPunishment;
import com.ikov.world.content.transportation.TeleportHandler;
import com.ikov.world.content.transportation.TeleportType;
import com.ikov.world.entity.impl.player.Player;
import com.ikov.world.entity.impl.player.PlayerSaving;

public class ModeratorCommands {

	public static void init() {
		CommandHandler.submit(new ModeratorCommand("tele") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String[] command = input.split(" ");
				int x = Integer.valueOf(command[0]), y = Integer.valueOf(command[1]);
				int z = player.getPosition().getZ();
				if (command.length > 2)
					z = Integer.valueOf(command[2]);
				Position position = new Position(x, y, z);
				player.moveTo(position);
				player.getPacketSender().sendMessage("Teleporting to " + position.toString());
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("teletome") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				Player player2 = World.getPlayerByName(playerName);
				if (player2 == null) {
					player.getPacketSender().sendMessage("Player " + playerName + " not found.");
					return false;
				}
				if (player2.getLocation() == Location.DUNGEONEERING) {
					player.getPacketSender().sendMessage("You cannot teleport a player out of dung.");
					return false;
				}
				if (player.getLocation() == Location.WILDERNESS) {
					player.getPacketSender().sendMessage("You cannot teleport a player into the wild.");
					return false;
				}
				if (player2.getLocation() == Location.DUEL_ARENA) {
					player.getPacketSender().sendMessage("You cannot do this to someone in duel arena.");
					return false;
				}
				boolean canTele = TeleportHandler.checkReqs(player, player2.getPosition().copy()) && player.getRegionInstance() == null && player2.getRegionInstance() == null;
				if (canTele) {
					TeleportHandler.teleportPlayer(player2, player.getPosition().copy(), TeleportType.NORMAL);
					player.getPacketSender().sendMessage("Teleporting player to you: " + player2.getUsername() + "");
					player2.getPacketSender().sendMessage("You're being teleported to " + player.getUsername() + "...");
				} else
					player.getPacketSender().sendMessage("You can not teleport that player at the moment. Maybe you or they are in a minigame?");
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("silenceyell") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				Player punishee = World.getPlayerByName(playerName);
				if (punishee != null) {
					punishee.setYellMute(true);
					punishee.getPacketSender().sendMessage("You have been yell muted! Please appeal on the forums.");
					player.getPacketSender().sendMessage("Player " + punishee.getUsername() + " was successfully muted!");
				} else {
					player.getPacketSender().sendMessage("Player " + playerName + " not found.");
					return false;
				}
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("unsilenceyell") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				Player punishee = World.getPlayerByName(playerName);
				if (punishee != null) {
					punishee.setYellMute(false);
					punishee.getPacketSender().sendMessage("You have been granted your yell ability again.");
					player.getPacketSender().sendMessage("Player " + playerName + " was successfully unmuted!");
				} else {
					player.getPacketSender().sendMessage("Player " + playerName + " not found.");
					return false;
				}
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("banvote") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				if (!PlayerSaving.playerExists(playerName)) {
					player.getPacketSender().sendMessage("Player " + playerName + " does not exist.");
					return false;
				} else {
					if (PlayerPunishment.isVoteBanned(playerName)) {
						player.getPacketSender().sendMessage("Player " + playerName + " already has an active vote ban.");
						return false;
					}
					Player other = World.getPlayerByName(playerName);
					PlayerPunishment.voteBan(playerName);
					other.getPacketSender().sendMessage("You have been banned from voting.");
					player.getPacketSender().sendMessage("You have banned " + other.getUsername() + " from voting.");
				}
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("unbanvote") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				if (!PlayerSaving.playerExists(playerName)) {
					player.getPacketSender().sendMessage("Player " + playerName + " does not exist.");
					return false;
				} else {
					if (!PlayerPunishment.isVoteBanned(playerName)) {
						player.getPacketSender().sendMessage("Player " + playerName + " is not vote banned.");
						return false;
					}
					Player other = World.getPlayerByName(playerName);
					PlayerPunishment.unVoteBan(playerName);
					other.getPacketSender().sendMessage("You have been unbanned from voting.");
					player.getPacketSender().sendMessage("You have unbanned " + other.getUsername() + " from voting.");
				}
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("ban") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				if (!PlayerSaving.playerExists(playerName)) {
					player.getPacketSender().sendMessage("Player " + playerName + " does not exist.");
					return false;
				} else {
					if (PlayerPunishment.isPlayerBanned(playerName)) {
						player.getPacketSender().sendMessage("Player " + playerName + " already has an active ban.");
						return false;
					}
					Player other = World.getPlayerByName(playerName);
					PlayerPunishment.ban(playerName);
					if (other != null) {
						World.deregister(other);
					}
					player.getPacketSender().sendMessage("Player " + playerName + " was successfully banned!");
				}
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("unban") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				if (!PlayerSaving.playerExists(playerName)) {
					player.getPacketSender().sendMessage("Player " + playerName + " does not exist.");
					return false;
				} else {
					if (!PlayerPunishment.isPlayerBanned(playerName)) {
						player.getPacketSender().sendMessage("Player " + playerName + " is not banned.");
						return false;
					}
					PlayerPunishment.unBan(playerName);
					player.getPacketSender().sendMessage("Player " + playerName + " was successfully unbanned.");
				}
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("ipmute") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				if (!PlayerSaving.playerExists(playerName)) {
					player.getPacketSender().sendMessage("Player " + playerName + " does not exist.");
					return false;
				} else {
					if (PlayerPunishment.isIpMuted(playerName)) {
						player.getPacketSender().sendMessage("Player " + playerName + " already has an active ip mute.");
						return false;
					}
					Player other = World.getPlayerByName(playerName);
					PlayerPunishment.ipMute(playerName);
					player.getPacketSender().sendMessage("Player " + playerName + " was successfully ip muted!");
					other.getPacketSender().sendMessage("You have been ip muted! Please appeal on the forums.");
				}
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("unipmute") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				if (!PlayerSaving.playerExists(playerName)) {
					player.getPacketSender().sendMessage("Player " + playerName + " does not exist.");
					return false;
				} else {
					if (!PlayerPunishment.isIpMuted(playerName)) {
						player.getPacketSender().sendMessage("Player " + playerName + " does not have an active ip mute!");
						return false;
					}
					Player other = World.getPlayerByName(playerName);
					PlayerPunishment.unIpMute(playerName);
					player.getPacketSender().sendMessage("Player " + playerName + " was successfully unipmuted!");
					other.getPacketSender().sendMessage("You have been unipmuted!");
				}
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("unmute") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				if (!PlayerSaving.playerExists(playerName)) {
					player.getPacketSender().sendMessage("Player " + playerName + " does not exist.");
					return false;
				} else {
					if (!PlayerPunishment.isMuted(playerName)) {
						player.getPacketSender().sendMessage("Player " + playerName + " is not muted.");
						return false;
					}
					Player other = World.getPlayerByName(playerName);
					PlayerPunishment.unMute(playerName);
					player.getPacketSender().sendMessage("Player " + playerName + " was successfully unmuted!");
					other.getPacketSender().sendMessage("You have been unmuted!");
				}
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("ipban") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				if (!PlayerSaving.playerExists(playerName)) {
					player.getPacketSender().sendMessage("Player " + playerName + " does not exist.");
					return false;
				} else {
					Player other = World.getPlayerByName(playerName);
					String ip;
					if (other == null) {
						ip = PlayerPunishment.getLastIpAddress(playerName);
					} else {
						ip = other.getHostAddress();
					}
					if (PlayerPunishment.isIpBanned(ip)) {
						player.getPacketSender().sendMessage("Player " + playerName + " already has an active ip ban on " + ip + ".");
						return false;
					}
					PlayerPunishment.ipBan(ip);
					PlayerPunishment.ban(playerName);
					if (other != null) {
						World.deregister(other);
					}
					player.getPacketSender().sendMessage("Player " + playerName + " was successfully banned on ip " + ip + "!");
				}
				return true;
			}

		});
		CommandHandler.submit(new ModeratorCommand("unipban") {

			@Override
			public boolean execute(Player player, String key, String input) throws Exception {
				String playerName = input;
				if (!PlayerSaving.playerExists(playerName)) {
					player.getPacketSender().sendMessage("Player " + playerName + " does not exist.");
					return false;
				} else {
					Player other = World.getPlayerByName(playerName);
					String ip;
					if (other == null) {
						ip = PlayerPunishment.getLastIpAddress(playerName);
					} else {
						ip = other.getHostAddress();
					}
					if (!PlayerPunishment.isIpBanned(ip)) {
						player.getPacketSender().sendMessage("Player " + playerName + " does not have an active ip ban on " + ip + ".");
						return false;
					}
					PlayerPunishment.unIpBan(ip);
					player.getPacketSender().sendMessage("Player " + playerName + " was successfully unipbanned on ip " + ip + "!");
				}
				return true;
			}

		});
	}

}

File: src/org/scripts/kotlin/content/commands/staff/ModeratorCommands.kt
package org.scripts.kotlin.content.commands.staff

import com.runelive.model.Locations
import com.runelive.util.Misc
import com.runelive.world.World
import com.runelive.world.content.PlayerPunishment
import com.runelive.world.content.transportation.TeleportHandler
import com.runelive.world.content.transportation.TeleportType
import com.runelive.world.entity.impl.player.Player
import com.runelive.world.entity.impl.player.PlayerSaving

/**
 * Created by Dave on 10/07/2016.
 */
class ModeratorCommands {

    object  Moderators {

        /**
         * @Author Jonathan Sirens Initiates Command
         */

       @JvmStatic fun initiate_command(player: Player, command: Array<String>, wholeCommand: String) {
            if (command[0].equals("massban", ignoreCase = true)) {
                val victimUsername = wholeCommand.substring(8)
                PlayerSaving.accountExists(victimUsername) { rs ->
                    if (rs.next()) {
                        // account exists
                        val other = World.getPlayerByName(victimUsername)
                        if (other == null) {
                            PlayerPunishment.massBan(player, victimUsername, Player(null))
                        } else {
                            val address = other.macAddress.toString()
                            val ip = other.hostAddress
                            PlayerPunishment.ban(victimUsername)
                            PlayerPunishment.pcBan(address)
                            PlayerPunishment.ipBan(ip)
                            World.deregister(other)
                            player.packetSender.sendMessage("You successfully mass banned '$victimUsername'.")
                        }
                    } else {
                        player.packetSender.sendMessage("Player $victimUsername does not exist.")
                    }
                }
            }
            if (command[0] == "gold") {
                val p = World.getPlayerByName(wholeCommand.substring(5))
                if (p != null) {
                    var gold: Long = 0
                    for (item in p.inventory.items) {
                        if (item != null && item.id > 0 && item.tradeable())
                            gold += item.definition.value.toLong()
                    }
                    for (item in p.equipment.items) {
                        if (item != null && item.id > 0 && item.tradeable())
                            gold += item.definition.value.toLong()
                    }
                    for (i in 0..8) {
                        for (item in p.getBank(i).items) {
                            if (item != null && item.id > 0 && item.tradeable())
                                gold += item.definition.value.toLong()
                        }
                    }
                    gold += p.moneyInPouch
                    player.packetSender.sendMessage(
                            p.username + " has " + Misc.insertCommasToNumber(gold.toString()) + " coins.")
                } else
                    player.packetSender.sendMessage("Can not find player online.")
            }
            if (command[0].equals("unmassban", ignoreCase = true)) {
                val victimUsername = wholeCommand.substring(10)
                PlayerSaving.accountExists(victimUsername) { rs ->
                    if (rs.next()) {
                        // account exists
                        val other = World.getPlayerByName(victimUsername)
                        if (other == null) {
                            PlayerPunishment.unmassBan(player, victimUsername, Player(null))
                        } else {
                            val address = other.macAddress.toString()
                            val ip = other.hostAddress
                            PlayerPunishment.unBan(victimUsername)
                            PlayerPunishment.unPcBan(address)
                            PlayerPunishment.unIpBan(ip)
                            World.deregister(other)
                            player.packetSender.sendMessage("You successfully unmass banned '$victimUsername'.")
                        }
                    } else {
                        player.packetSender.sendMessage("Player $victimUsername does not exist.")
                    }
                }
            }
            if (command[0].equals("teletome", ignoreCase = true)) {
                val playerToTele = wholeCommand.substring(9)
                val player2 = World.getPlayerByName(playerToTele)
                if (World.getPlayerByName(playerToTele)!!.location === Locations.Location.DUEL_ARENA) {
                    player.packetSender.sendMessage("Why are you trying to move a player out of duel arena?")
                    return
                }
                if (player2.location === Locations.Location.DUNGEONEERING) {
                    player.packetSender.sendMessage("You cannot teleport a player out of dung?")
                    return
                }
                if (player.location === Locations.Location.WILDERNESS) {
                    player.packetSender.sendMessage("You cannot teleport a player into the wild... What're you thinking?")
                    return
                }
                if (player2.location === Locations.Location.DUEL_ARENA) {
                    player.packetSender.sendMessage("You cannot do this to someone in duel arena.")
                    return
                }
                val canTele = TeleportHandler.checkReqs(player, player2.position.copy())
                        && player.regionInstance == null && player2.regionInstance == null
                if (canTele) {
                    TeleportHandler.teleportPlayer(player2, player.position.copy(), TeleportType.NORMAL)
                    player.packetSender.sendMessage("Teleporting player to you: " + player2.username + "")
                    player2.packetSender.sendMessage("You're being teleported to " + player.username + "...")
                } else
                    player.packetSender.sendMessage(
                            "You can not teleport that player at the moment. Maybe you or they are in a minigame?")
            }
            if (wholeCommand.startsWith("silenceyell")) {
                val yellmute = wholeCommand.substring(12)
                val punishee = World.getPlayerByName(yellmute)
                PlayerSaving.accountExists(yellmute) { rs ->
                    if (rs.next()) {
                        // account exists
                        punishee.isYellMute = true
                        punishee.packetSender.sendMessage("You have been yell muted! Please appeal on the forums.")
                        player.packetSender.sendMessage("Player " + punishee.username + " was successfully muted!")
                    } else {
                        player.packetSender.sendMessage("Player $yellmute does not exist.")
                    }
                }
            }
            if (wholeCommand.startsWith("unsilenceyell")) {
                val yellmute = wholeCommand.substring(14)
                val punishee = World.getPlayerByName(yellmute)
                PlayerSaving.accountExists(yellmute) { rs ->
                    if (rs.next()) {
                        // account exists
                        punishee.isYellMute = false
                        punishee.packetSender.sendMessage("You have been granted your yell ability again.")
                        player.packetSender.sendMessage("Player " + punishee.username + " was successfully unmuted!")
                    } else {
                        player.packetSender.sendMessage("Player $yellmute does not exist.")
                    }
                }
            }
            if (command[0].equals("ban", ignoreCase = true)) {
                val ban_player = wholeCommand.substring(4)
                PlayerSaving.accountExists(ban_player) { rs ->
                    if (rs.next()) {
                        // account exists
                        if (PlayerPunishment.isPlayerBanned(ban_player)) {
                            player.packetSender.sendMessage("Player $ban_player already has an active ban.")
                            return@accountExists
                        }
                        val other = World.getPlayerByName(ban_player)
                        PlayerPunishment.ban(ban_player)
                        if (other != null) {
                            World.deregister(other)
                        }
                        player.packetSender.sendMessage("Player $ban_player was successfully banned!")
                    } else {
                        player.packetSender.sendMessage("Player $ban_player does not exist.")
                    }
                }
            }
            if (command[0].equals("ipmute", ignoreCase = true)) {
                val mute_player = wholeCommand.substring(7)
                PlayerSaving.accountExists(mute_player) { rs ->
                    if (rs.next()) {
                        // account exists
                        if (PlayerPunishment.isIpMuted(mute_player)) {
                            player.packetSender.sendMessage("Player $mute_player already has an active ip mute.")
                            return@accountExists
                        }
                        val other = World.getPlayerByName(mute_player)
                        PlayerPunishment.ipMute(mute_player)
                        player.packetSender.sendMessage("Player $mute_player was successfully ip muted!")
                        other.packetSender.sendMessage("You have been ip muted! Please appeal on the forums.")
                    } else {
                        player.packetSender.sendMessage("Player $mute_player does not exist.")
                    }
                }
            }
            if (command[0].equals("unipmute", ignoreCase = true)) {
                val mute_player = wholeCommand.substring(9)
                PlayerSaving.accountExists(mute_player) { rs ->
                    if (rs.next()) {
                        // account exists
                        if (!PlayerPunishment.isIpMuted(mute_player)) {
                            player.packetSender.sendMessage("Player $mute_player does not have an active ip mute!")
                            return@accountExists
                        }
                        val other = World.getPlayerByName(mute_player)
                        PlayerPunishment.unIpMute(mute_player)
                        player.packetSender.sendMessage("Player $mute_player was successfully unipmuted!")
                        other.packetSender.sendMessage("You have been unipmuted!")
                    } else {
                        player.packetSender.sendMessage("Player $mute_player does not exist.")
                    }
                }
            }
            if (command[0].equals("unban", ignoreCase = true)) {
                val ban_player = wholeCommand.substring(6)
                PlayerSaving.accountExists(ban_player) { rs ->
                    if (rs.next()) {
                        // account exists
                        if (!PlayerPunishment.isPlayerBanned(ban_player)) {
                            player.packetSender.sendMessage("Player $ban_player is not banned.")
                            return@accountExists
                        }
                        PlayerPunishment.unBan(ban_player)
                        player.packetSender.sendMessage("Player $ban_player was successfully unbanned.")
                    } else {
                        player.packetSender.sendMessage("Player $ban_player does not exist.")
                    }
                }
            }
            if (command[0] == "unbanvote") {
                val vote_player = wholeCommand.substring(10)
                PlayerSaving.accountExists(vote_player) { rs ->
                    if (rs.next()) {
                        // account exists
                        if (!PlayerPunishment.isVoteBanned(vote_player)) {
                            player.packetSender.sendMessage("Player $vote_player is not vote banned.")
                            return@accountExists
                        }
                        val other = World.getPlayerByName(vote_player)
                        PlayerPunishment.unVoteBan(vote_player)
                        other.packetSender.sendMessage("You have been unbanned from voting.")
                        player.packetSender.sendMessage("You have unbanned " + other.username + " from voting.")
                    } else {
                        player.packetSender.sendMessage("Player $vote_player does not exist.")
                    }
                }
            }
            if (command[0].equals("banvote", ignoreCase = true)) {
                val vote_player = wholeCommand.substring(8)
                PlayerSaving.accountExists(vote_player) { rs ->
                    if (rs.next()) {
                        // account exists
                        if (PlayerPunishment.isVoteBanned(vote_player)) {
                            player.packetSender.sendMessage("Player $vote_player already has an active vote ban.")
                            return@accountExists
                        }
                        val other = World.getPlayerByName(vote_player)
                        PlayerPunishment.voteBan(vote_player)
                        other.packetSender.sendMessage("You have been banned from voting.")
                        player.packetSender.sendMessage("You have banned " + other.username + " from voting.")
                    } else {
                        player.packetSender.sendMessage("Player $vote_player does not exist.")
                    }
                }
            }
        }
    }

    object GlobalModerators {

        @JvmStatic fun initiate_command(player: Player, command: Array<String>, wholeCommand: String) {

            }
        }
    }
